astropy/wcs/tests/test_utils.py

```diff
@@ -473,6 +534,44 @@ class FakeFrame(BaseCoordinateFrame):
     assert mywcs.wcs.dateobs == Time("J2000").utc.fits
 
 
+def test_body_to_wcs_frame():
+    # Import astropy.coordinates here to avoid circular imports
+    from astropy.coordinates import BaseCoordinateFrame
+
+    class IAUMARSSphereFrame(BaseCoordinateFrame):
+        name = "Mars"
+        representation_type = SphericalRepresentation
+
+    frame = IAUMARSSphereFrame()
```

Comment:
Maybe again do the failures at the end?

Reply:
not sure my understanding is correct here: I have moved the test on the error after the definitions of the classes.

Reply:
No, that is not what I meant: I just meant that in this function I feel it is better first to test things as they are supposed to work, and then at the end test that the right errors are raised. I.e., move this `pytest.raises` to the very end of the test function.

Reply:
Understood, sorry.